model/src/test/kotlin/utils/LicenseResolverTest.kt

```diff
@@ -0,0 +1,261 @@
+                setupScanResult(
+                    id, ProvenanceType.VCS, listOf(
+                        LicenseFinding(
+                            license = "Apache-2.0",
+                            location = TextLocation("a.txt", startLine = 1, endLine = 1)
+                        ),
+                        LicenseFinding(
+                            license = "MIT",
+                            location = TextLocation("m.txt", startLine = 1, endLine = 1)
+                        )
+                    )
+                )
+
+                val result = createResolver().getDetectedLicensesForId(id)
+
+                result should containExactlyInAnyOrder("Apache-2.0", "MIT")
+            }
+        }
+
+        "Given a package with a scan result for source artifact, getDetectedLicensesForId()" should {
+            "return all detected licenses" {
+                val id = setupPackage().id
+                setupScanResult(
+                    id, ProvenanceType.SOURCE_ARTIFACT, listOf(
+                        LicenseFinding(
+                            license = "Apache-2.0",
+                            location = TextLocation("some/path", startLine = 1, endLine = 1)
+                        ),
+                        LicenseFinding(
+                            license = "MIT",
+                            location = TextLocation("some/path", startLine = 1, endLine = 1)
+                        )
+                    )
+                )
+
+                val result = createResolver().getDetectedLicensesForId(id)
+
+                result should containExactlyInAnyOrder("Apache-2.0", "MIT")
+            }
+        }
+
+        "Given a package with a scan result for VCS and source artifact, getDetectedLicensesForId()" should {
+            "return all detected licenses" {
+                val id = setupPackage().id
+                setupScanResult(
+                    id, ProvenanceType.VCS, listOf(
+                        LicenseFinding(
+                            license = "Apache-2.0",
+                            location = TextLocation("some/path", startLine = 1, endLine = 1)
+                        )
+                    )
+                )
+                setupScanResult(
+                    id, ProvenanceType.SOURCE_ARTIFACT, listOf(
+                        LicenseFinding(
+                            license = "MIT",
+                            location = TextLocation("some/path", startLine = 1, endLine = 1)
+                        )
+                    )
+                )
+
+                val result = createResolver().getDetectedLicensesForId(id)
+
+                result should containExactlyInAnyOrder("Apache-2.0", "MIT")
+            }
+        }
+
+        "Given a package with multiple scan results for VCS and source artifact, getDetectedLicensesForId()" should {
+            "return all detected licenses" {
+                val id = setupPackage().id
+                setupScanResult(
```

Comment:
Ups, this test should have contained 2 scan results for each, VCS and source artifact. let me update it...